trace/trace_test.go

```diff
@@ -0,0 +1,199 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package trace
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"k8s.io/klog"
+)
+
+func TestStep(t *testing.T) {
+	tests := map[string]*struct {
```

Comment:
IMO, I think something like this makes the test table more readable:
```go
tests := []struct{
    name string
    inputString string
    expectedTrace *Trace
}{
      {
            name: "string is empty",
            inputString: "",
            expectedTrace: &Trace{....}
      } 
}

for _, test := range tests {
    t.Run(test.Name, t.Run(t *testing.T) {
         ....
    })
}
```

With maps it's not immediately clear that the key value is the description of the tests. What do you think?

Reply:
nice we can do it

Reply:
updated